src/test/java/com/epam/page/object/generator/integrationalTests/MainIntegrationTest.java

```diff
@@ -0,0 +1,401 @@
+                                         boolean checkLocatorUniqueness,
+                                         boolean forceGenerateFiles) throws IOException {
+        List<String> urls = new ArrayList<>();
+        urls.add(url);
+        SupportedTypesContainer bc = new SupportedTypesContainer();
+        JsonRuleMapper parser = new JsonRuleMapper(new File(jsonPath), new ObjectMapper());
+        XpathToCssTransformation xpathToCssTransformation = new XpathToCssTransformation();
+        JavaPoetAdapter javaPoetAdapter = new JavaPoetAdapter(bc, xpathToCssTransformation);
+
+        ValidatorsStarter validatorsStarter = new ValidatorsStarter(bc);
+        validatorsStarter.setCheckLocatorsUniqueness(checkLocatorUniqueness);
+
+        PageObjectsGenerator pog = new PageObjectsGenerator(parser, validatorsStarter,
+            javaPoetAdapter, RESOURCE_DIR, urls, PACKAGE_TEST_NAME);
+        pog.setForceGenerateFile(forceGenerateFiles);
+        return pog;
+    }
+
+    public MainIntegrationTest(List<TestClassesDTO> list) {
+        this.caseClassesList = list;
+    }
+
+    /**
+     * Forms List<Object[]> which contains parameters for testing. Every list's element
+     * (it's Object array) contains one parameter - list of TestClassesDTO objects. This list
+     * represents each test case. For this purpose data() method scans
+     * through scr/test/resources/manual to find all test cases dirs and then form list
+     * of TestClassesDTO objects for every test case
+     *
+     * About regex: .+\.[a-z]+$
+     * This regex is used for matching files with extensions (in this particular situation for
+     * matching .java or .properties files). Example:
+     * src/test/resources/manual/google/Google.java - matches
+     * src/test/resources/manual/google             - doesn't match
+     *
+     * @return list of parameters set for every test case
+     */
+    @Parameters
+    public static Iterable<Object[]> data() throws IOException, ClassNotFoundException {
+
+        List<Path> caseDirs = getCaseDirsNames(
+            RESOURCE_DIR + MANUAL_DIR,
+            MAX_DEPTH_FOR_DIRS
+        );
+        List<Path> insideCaseDirs;
+        List<Path> javaFilesList;
+        List<Path> casePropertiesList;
+        List<Object[]> testParameters = new ArrayList<>();
+        Properties caseProperties;
+        List<TestClassesDTO> objectList;
+
+        for (Path caseDir : caseDirs) {
+            objectList = new ArrayList<>();
+            caseProperties = new Properties();
+
+            insideCaseDirs = Files.find(
+                caseDir,
+                MAX_DEPTH_FOR_DIRS,
+                (path, basicFileAttributes) -> !FileSystems
+                    .getDefault()
+                    .getPathMatcher("regex:.+\\.[a-z]+$")
+                    .matches(path) && !path.equals(caseDir)
+            ).collect(Collectors.toList());
+
+            casePropertiesList = Files.find(
```

Comment:
Same problem here. Much better (and safe) approach:
            Optional<Path> propertiesFilePath = Files.list(caseDir)
                    .filter(p -> Files.isRegularFile(p))
                    .filter(p -> p.toString().endsWith(".properties"))
                    .findAny();
            if (propertiesFilePath.isPresent()) {
                caseProperties.load(Files.newInputStream(propertiesFilePath.get()));
            }

Reply:
Forgot to add: Input stream should be closed